direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -330,15 +335,14 @@ private void submitConsumerWithObserver(
         () -> {
           handle.set(createObserveHandle(shutdown, seekOffsets, consumer, latch));
           final AtomicReference<KafkaConsumer<Object, Object>> consumerRef;
-          final AtomicReference<VectorClock> clock = new AtomicReference<>(VectorClock.of(1));
-          final Map<Integer, Integer> partitionToClockDimension = new ConcurrentHashMap<>();
+          final AtomicReference<PartitionedWatermarkEstimator> watermarkEstimator =
+              new AtomicReference<>(PartitionedWatermarkEstimator.empty());
```

Comment:
Please leave this empty (`new AtomicReference<>()`)

Reply:
Actually I had it empty but when I did dev test program was failing on NullPointerException:
```
 if (poll.isEmpty()) {
       consumer.onIdle(watermarkEstimator.get());
  }
```
It seems that lot of code is running also when no partitions are assigned yet.

Reply:
KafkaLogReader requires refactoring and handle that properly, but it should be part of another issue.

Reply:
How complex this refactor can be? If we need a default estimator, we can use something like `() -> Long.MIN_VALUE`, that would feel a lot safer.

Reply:
Will be addressed in O2-Czech-Republic#155